pithos/plugins/notify.py

```diff
@@ -77,3 +79,47 @@ def on_disable(self):
         if self._shutdown_handler:
             self._app.disconnect(self._shutdown_handler)
             self._shutdown_handler = 0
+
+
+class NotifyPluginPrefsDialog(Gtk.Dialog):
+    __gtype_name__ = 'NotifyPluginPrefsDialog'
+
+    def __init__(self, window, settings):
+        super().__init__(use_header_bar=1)
+        self.set_title(_('Notification Preferences'))
+        self.set_default_size(300, -1)
+        self.set_resizable(False)
+        self.connect('delete-event', self.on_close)
+
+        self.pithos = window
+        self.settings = settings
+        self.show_album = self.settings['data'] == 'True' if self.settings['data'] else False
+
+        box = Gtk.Box()
+        label = Gtk.Label()
+        label.set_markup('<b>{}</b>\n{}'.format(_('Show Album Info'), None))
+        label.set_halign(Gtk.Align.START)
+        box.pack_start(label, True, True, 4)
+
+        self.switch = Gtk.Switch()
+        self.switch.connect('notify::active', self.toggle_album_info)
+        self.switch.set_active(self.show_album)
+        self.settings.connect('changed::enabled', self._on_album_info_plugin_enabled)
+        self.switch.set_halign(Gtk.Align.END)
+        self.switch.set_valign(Gtk.Align.CENTER)
+        box.pack_end(self.switch, False, False, 2)
+
+        content_area = self.get_content_area()
```

Comment:
This is deprecated, can you just do `self.add(box)`? I forget if that works if you could test.

Reply:
```(pithos:2): Gtk-WARNING **: 13:00:25.803: Attempting to add a widget with type GtkBox to a NotifyPluginPrefsDialog, but as a GtkBin subclass a NotifyPluginPrefsDialog can only contain one widget at a time; it already contains a widget of type GtkBox```

I'll test with it though.

Reply:
The variable naming is different but the latest (assuming it's up to date since it was last edited over two years ago) [example](https://python-gtk-3-tutorial.readthedocs.io/en/latest/dialogs.html#example) suggests the structure is correct.

Reply:
I see the deprecation notice with GTK4. I've switched it from a GtkDialog to a GtkWindow object and that seems to be working fine minus one issue where the preference is displayed away from the plugins and main application windows. I'll keep looking to see if I can resolve this.

Reply:
I think this is just a byproduct of using Gtk.Window instead of Gtk.Dialog. Window managers will position the Window object the same as other window objects. The other preferences are still using Gtk.Dialog so this will create a bit of inconsistency in terms of user experience.

I'll commit whichever you prefer. If it's Gtk.Window, this is ready to merge as is. If it is Gtk.Dialog, I can push the previous state.

Reply:
I thought of another thing. If Gtk.Window is the direction you want things to go while also wanting the user experience to be uniform, I can work on PRs for all of the other plugins with preferences. Then the PRs could be review/merged simultaneously.